packages/eas-shared/src/run/ios/simulator.ts

```diff
@@ -270,27 +270,26 @@ async function getClientForSDK(sdkVersionString?: string) {
   };
 }
 
-export async function expoVersionOnSimulatorAsync(udid: string): Promise<string | null> {
-  const localPath = await getContainerPathAsync({
-    udid,
-    bundleIdentifier: EXPO_GO_BUNDLE_IDENTIFIER,
-  });
-  if (!localPath) {
-    return null;
-  }
+export async function expoSDKSupportedVersionsOnSimulatorAsync(
+  udid: string
+): Promise<string[] | null> {
+  try {
+    const localPath = await getContainerPathAsync({
+      udid,
+      bundleIdentifier: EXPO_GO_BUNDLE_IDENTIFIER,
+    });
+    if (!localPath) {
+      return null;
+    }
 
-  const regex = /Exponent-([0-9.]+).*\.app$/;
-  const regexMatch = regex.exec(localPath);
-  if (!regexMatch) {
-    return null;
-  }
+    const builtInfoPlistPath = path.join(localPath, 'EXSDKVersions.plist');
```

Comment:
Unfortunately this is only possible on simulators, I tried using the HouseArrest Service to retrieve the EXSDKVersions on a real device but that doesn't work either given that VendContainer is only supported on dev-signed apps